NumLib/DOF/LocalToGlobalIndexMap.h

```diff
@@ -133,6 +133,14 @@ class LocalToGlobalIndexMap final
         std::size_t const mesh_id,
         const unsigned component_id, const unsigned comp_id_write);
 
+    /// The global component id for the specific variable (like velocity) and a
+    /// component (like x, or y, or z).
+    std::size_t global_component(int const variable_id,
+                                 int const component_id) const
+    {
+        return _variable_component_offsets[variable_id] + component_id;
```

Comment:
There are two ways add components to global vectors, i.e.  BY_COMPONENT, or  BY_LOCATION.  It sounds that this is only for  BY_COMPONENT.